mypy/plugin.py

```diff
@@ -379,3 +421,332 @@ def int_pow_callback(ctx: MethodContext) -> Type:
+        """Get the CallableType that's attached to a CallExpr."""
+        callable_type = None
+        if (isinstance(call, CallExpr)
+                and isinstance(call.callee, RefExpr)
+                and isinstance(call.callee.node, Var)
+                and call.callee.node.type):
+            callee_type = call.callee.node.type
+            if isinstance(callee_type, Overloaded):
+                # We take the last overload.
+                callable_type = callee_type.items()[-1]
+            elif isinstance(callee_type, CallableType):
+                callable_type = callee_type
+        return callable_type
+
+    def get_argument(call: CallExpr, name: str) -> Optional[Expression]:
+        """Return the expression for the specific argument."""
+        callee = get_callable_type(call)
+        if not callee:
+            return None
+
+        argument = callee.argument_by_name(name)
+        if not argument:
+            return None
+
+        if not argument.name and not argument.pos:
+            return None
+
+        for i, (attr_name, attr_value) in enumerate(zip(call.arg_names, call.args)):
+            if argument.pos is not None and not attr_name and i == argument.pos:
+                return attr_value
+            if attr_name == argument.name:
+                return attr_value
+        return None
+
+    def get_bool_argument(expr: Expression, name: str, default: bool) -> bool:
+        """Return the value of an argument name in the give Expression.
+
+        If it's a CallExpr and the argument is one of the args then return it.
+        Otherwise return the default value for the argument.
+        """
+        if isinstance(expr, CallExpr):
+            attr_value = get_argument(expr, name)
+            if attr_value:
+                ret = ctx.api.parse_bool(attr_value)
+                if ret is None:
+                    ctx.api.fail('"{}" argument must be True or False.'.format(name), expr)
+                    return default
+                return ret
+        return default
+
+    def is_class_var(expr: NameExpr) -> bool:
+        """Return whether the expression is ClassVar[...]"""
+        if isinstance(expr.node, Var):
+            return expr.node.is_classvar
+        return False
+
+    decorator = ctx.reason
+
+    # auto_attribs means we also generate Attributes from annotated variables.
+    auto_attribs = get_bool_argument(decorator, 'auto_attribs', auto_attribs_default)
+
+    def get_attributes_for_body(body: List[Statement]) -> List[Attribute]:
+        attributes = OrderedDict()  # type: OrderedDict[str, Attribute]
+
+        for stmt in body:
+            if isinstance(stmt, AssignmentStmt) and isinstance(stmt.lvalues[0], NameExpr):
+                lhs = stmt.lvalues[0]
+                assert isinstance(lhs, NameExpr)
+                name = lhs.name
+                typ = stmt.type
+
+                func_name = called_function(stmt.rvalue)
```

Comment:
``called_function`` is used only once here, doesn't it make sense to just inline it?